spark/src/test/scala/org/apache/spark/sql/delta/DeltaTypeWideningAutomaticSuite.scala

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.{DataFrame, QueryTest, SaveMode}
+import org.apache.spark.sql.functions.{col, lit}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.internal.SQLConf.StoreAssignmentPolicy
+import org.apache.spark.sql.types._
+
+
+/**
+ * Suite covering widening columns and fields type as part of automatic schema evolution when the
+ * type widening table feature is supported.
+ */
+class DeltaTypeWideningAutomaticSuite
```

Comment:
Sorry to do this but it probably still needs to say ..AutomaticTypeEvolutionSuite. Otherwise it's not clear what an AutomaticSuite is.

Reply:
👍 I kept the prefix `DeltaTypeWidening` so that type widening test suites are grouped together